org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/debug/AbstractDebugAdapterLaunchShortcut.java

```diff
@@ -45,20 +45,30 @@
 
 public abstract class AbstractDebugAdapterLaunchShortcut implements ILaunchShortcut2 {
 
-	private final String contentTypeId;
+	private final String[] contentTypeIds;
 	private final ILaunchManager launchManager = DebugPlugin.getDefault().getLaunchManager();
 	private final ILaunchConfigurationType configType;
 	private final boolean autoStartNewlyCreatedConfiguration;
 
 	public AbstractDebugAdapterLaunchShortcut(String launchConfigTypeId, String contentTypeId, boolean autoStartNewlyCreatedConfiguration) {
+		this(launchConfigTypeId, new String[] {contentTypeId}, autoStartNewlyCreatedConfiguration);
+	}
+
+	public AbstractDebugAdapterLaunchShortcut(String launchConfigTypeId, String[] contentTypeIds, boolean autoStartNewlyCreatedConfiguration) {
 		this.autoStartNewlyCreatedConfiguration = autoStartNewlyCreatedConfiguration;
-		this.contentTypeId = contentTypeId;
+		this.contentTypeIds = contentTypeIds;
 		this.configType = launchManager.getLaunchConfigurationType(launchConfigTypeId);
 	}
 
 	public boolean canLaunch(File file) {
-		return file.exists()
-				&& Platform.getContentTypeManager().getContentType(contentTypeId).isAssociatedWith(file.getName());
+		if (file.exists()) {
+			for (String contentTypeId : contentTypeIds) {
```

Comment:
Just for the fun, feel free to use lambdas in that case
`Arrays.stream(contentTypeIds).map(Platform.getContentTypeManager::getContentTypes()).anyMatch(type -> type.isAssociatedWiht(file.getName())`.

Reply:
OK